tensorflow_encrypted/protocol/securenn.py

```diff
@@ -109,3 +118,54 @@ def max_pool(self, x):
 
     def dmax_pool_efficient(self, x):
         raise NotImplementedError
+
+
+def _lsb_private(prot: SecureNN, y: PondPrivateTensor):
+    with tf.name_scope('lsb_mask'):
+        with tf.device(prot.crypto_producer.device_name):
+            x = prot.tensor_factory.Tensor.sample_uniform(y.shape)
+            xbits = x.binarize()
+            xlsb = xbits[..., 0]
+            xlsb0, xlsb1 = prot._share(xlsb, p)
+            x = PondPrivateTensor(prot, *prot._share(x), is_scaled=True)
+            xbits = PondPrivateTensor(prot, *prot._share(xbits), is_scaled=False)
+            # TODO: Generate zero mask?
+
+        devices = [prot.server0.device_name, prot.server1.device_name]
+        bits_device = random.choice(devices)
+        with tf.device(bits_device):
+            b = _generate_random_bits(y.shape)
+            b0, b1 = b.unwrapped
+
+    with tf.name_scope('lsb'):
+        r = (y + x).reveal()
+        rbits0, rbits1 = prot.binarize(r).unwrapped
+        rlsb0, rlsb1 = rbits0[..., 0], rbits1[..., 0]
+
+        bp = prot.private_compare(xbits, r, b)
+        bp0, bp1 = bp.unwrapped
+
+        with tf.device(prot.server0.device_name):
+            gamma_on_0 = bp0 + b0 * bp0 * (-2)
```

Comment:
this is just an xor

Reply:
I don't think this is exactly the xor?  it looks like `xor(bp0, b0) - b0` to me.  Can replace to be that if needed, similar for below